FAIL: test_clustering_three_ensembles_two_identical on non-x86 64bit arches (aarch64, ppc64, ppc64le and s390x)

### Expected behaviour
All tests pass.

### Actual behaviour
```
Ran 4554 tests in 572.048s
FAILED (SKIP=16, failures=1)
```

### Code to reproduce the behaviour
On Fedora rawhide (27) running on [aarch64](https://kojipkgs.fedoraproject.org//work/tasks/5679/19915679/build.log), [ppc64](https://kojipkgs.fedoraproject.org//work/tasks/5773/19915773/build.log), [ppc64le](https://kojipkgs.fedoraproject.org//work/tasks/5782/19915782/build.log) and [s390x](https://kojipkgs.fedoraproject.org//work/tasks/5785/19915785/build.log), the following failed test can be observed (log is from aarc64 build):
```
======================================================================
FAIL: test_clustering_three_ensembles_two_identical (MDAnalysisTests.analysis.test_encore.TestEncoreClustering)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/MDAnalysis-0.16.1/MDAnalysisTests-0.16.1/MDAnalysisTests/analysis/test_encore.py", line 486, in test_clustering_three_ensembles_two_identical
    err_msg="Unexpected result: {0}".format(cluster_collection))
  File "/usr/lib64/python2.7/site-packages/numpy/testing/utils.py", line 416, in assert_equal
    raise AssertionError(msg)
AssertionError: 
Items are not equal:
Unexpected result: 0 (size:3,centroid:1): array([ 1, 21, 42])
1 (size:2,centroid:2): array([ 2, 43])
2 (size:2,centroid:3): array([ 3, 44])
3 (size:1,centroid:4): array([4])
4 (size:1,centroid:5): array([5])
5 (size:1,centroid:6): array([6])
6 (size:1,centroid:7): array([7])
7 (size:2,centroid:8): array([ 8, 49])
8 (size:2,centroid:9): array([ 9, 50])
9 (size:2,centroid:10): array([10, 51])
10 (size:1,centroid:11): array([11])
11 (size:1,centroid:12): array([12])
12 (size:2,centroid:13): array([13, 54])
13 (size:1,centroid:14): array([14])
14 (size:1,centroid:15): array([15])
15 (size:1,centroid:16): array([16])
16 (size:1,centroid:17): array([17])
17 (size:4,centroid:18): array([18, 38, 39, 59])
18 (size:3,centroid:23): array([22, 23, 24])
19 (size:2,centroid:26): array([25, 26])
20 (size:2,centroid:28): array([27, 28])
21 (size:3,centroid:30): array([29, 30, 31])
22 (size:3,centroid:33): array([32, 33, 34])
23 (size:3,centroid:36): array([35, 36, 37])
24 (size:3,centroid:41): array([ 0, 20, 41])
25 (size:1,centroid:45): array([45])
26 (size:1,centroid:46): array([46])
27 (size:1,centroid:47): array([47])
28 (size:1,centroid:48): array([48])
29 (size:1,centroid:52): array([52])
30 (size:1,centroid:53): array([53])
31 (size:1,centroid:55): array([55])
32 (size:1,centroid:56): array([56])
33 (size:1,centroid:57): array([57])
34 (size:1,centroid:58): array([58])
35 (size:3,centroid:60): array([19, 40, 60])
 ACTUAL: 36
 DESIRED: 40
-------------------- >> begin captured logging << --------------------
MDAnalysis.topology.PSF: DEBUG: PSF file /builddir/build/BUILD/MDAnalysis-0.16.1/MDAnalysisTests-0.16.1/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Atomids object at 0xffff58c49e50> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Atomnames object at 0xffff58c49e10> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Atomtypes object at 0xffff58c49dd0> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Charges object at 0xffff58c49d90> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Masses object at 0xffff58c49d50> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Resids object at 0xffff58c49fd0> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Resnums object at 0xffff58c49d10> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Resnames object at 0xffff58c49250> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Segids object at 0xffff58c49f90> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Atomindices object at 0xffff58c49c90> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Resindices object at 0xffff58c49c50> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Segindices object at 0xffff58c49c10> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Bonds object at 0xffff58c49b90> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Angles object at 0xffff58c49390> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Dihedrals object at 0xffff58c49290> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Impropers object at 0xffff58c49b50> to topology
MDAnalysis.core.universe: DEBUG: Universe.load_new(): loading [[[ 11.73604393   8.50079727 -10.44528103]
  [ 12.36511898   7.83993578 -10.83484173]
  [ 12.09194851   9.441535   -10.72461128]
  ..., 
  [  6.51260424  18.44701767  -7.13405323]
  [  6.30018616  19.36348534  -7.93591595]
  [  5.58540154  17.5896244   -6.96566153]]
 [[ 12.97540283   8.5564661   -8.79086781]
  [ 13.66876411   8.11343288  -8.09570789]
  [ 13.19270515   9.55035019  -9.03054714]
  ..., 
  [  7.34918356  17.4371357   -6.57104731]
  [  7.43909597  17.90647507  -7.75675392]
  [  6.21367168  17.11502075  -6.09452438]]
 [[ 13.04726887   7.63489389  -9.00248337]
  [ 13.95461655   7.6078577   -8.37104511]
  [ 13.05357552   8.57965088  -9.43249798]
  ..., 
  [  7.36733294  17.43116379  -6.58304739]
  [  7.42952394  18.0872364   -7.64184904]
  [  6.29180861  16.84422874  -6.30289555]]
 ..., 
 [[ 15.64185715   6.52680159  -7.78829479]
  [ 16.17948151   6.46231985  -6.86501741]
  [ 15.8962183    7.40947819  -8.22259808]
  ..., 
  [ 12.37435722  14.83569622  -6.58019352]
  [ 12.4016571   15.75725174  -7.4538641 ]
  [ 11.34606457  14.13633633  -6.49367189]]
 [[ 16.01750946   6.42328167  -8.06725407]
  [ 16.6257       6.14889956  -7.24784851]
  [ 16.26796532   7.37695169  -8.37005711]
  ..., 
  [ 12.80069256  15.77023315  -6.08095074]
  [ 12.70214462  16.68847656  -6.90102482]
  [ 12.32308102  14.67083359  -6.31280994]]
 [[ 16.29778099   6.83979559  -7.62298918]
  [ 16.82201767   6.56630898  -6.70722151]
  [ 16.76083183   7.6656146   -7.95306826]
  ..., 
  [ 12.63667011  15.56686878  -6.11850452]
  [ 12.8277998   16.21443558  -7.16725492]
  [ 11.55105019  14.87915421  -5.94013405]]]...
MDAnalysis.topology.PSF: DEBUG: PSF file /builddir/build/BUILD/MDAnalysis-0.16.1/MDAnalysisTests-0.16.1/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Atomids object at 0xffff58903190> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Atomnames object at 0xffff589031d0> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Atomtypes object at 0xffff58903210> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Charges object at 0xffff58903250> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Masses object at 0xffff58903290> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Resids object at 0xffff58903310> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Resnums object at 0xffff589032d0> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Resnames object at 0xffff58903350> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Segids object at 0xffff58903390> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Atomindices object at 0xffff58903450> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Resindices object at 0xffff58903490> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Segindices object at 0xffff589034d0> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Bonds object at 0xffff58903550> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Angles object at 0xffff58903590> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Dihedrals object at 0xffff589035d0> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Impropers object at 0xffff58903610> to topology
MDAnalysis.core.universe: DEBUG: Universe.load_new(): loading [[[ 26.87939835  53.00944901  40.99979401]
  [ 26.80416107  52.05094147  41.55620956]
  [ 26.58082199  53.76752472  41.59049988]
  ..., 
  [ 14.87982082  56.23511505  39.41582489]
  [ 14.59319496  57.44499207  39.77984619]
  [ 15.05454922  55.95846176  38.18389893]]
 [[ 27.79439926  52.48551559  40.48461914]
  [ 28.64513397  52.68635178  39.99812698]
  [ 27.86498642  51.63056564  41.1264534 ]
  ..., 
  [ 15.3743372   55.06335449  39.18051147]
  [ 15.21354675  56.26719666  39.45918274]
  [ 15.14764881  54.60814285  38.06523895]]
 [[ 28.26911163  51.67028046  41.10035706]
  [ 29.20409012  51.85968018  40.61678314]
  [ 28.36113548  50.66461563  41.48442078]
  ..., 
  [ 14.91166115  54.62375259  40.35363007]
  [ 14.22121239  55.64897919  40.60663223]
  [ 15.63880348  54.53017807  39.34329224]]
 ..., 
 [[ 26.54650688  52.46332169  40.16776276]
  [ 27.58373642  52.59495926  40.22349548]
  [ 26.40011978  51.45680618  40.39321136]
  ..., 
  [ 16.67232513  54.58612061  41.22190475]
  [ 16.0298481   55.61537552  41.42271805]
  [ 17.51662445  54.52005386  40.30110168]]
 [[ 25.87160873  52.20184708  40.38906479]
  [ 26.71255112  52.10401917  40.89827728]
  [ 25.50161743  51.15327454  40.24638748]
  ..., 
  [ 16.31178474  54.46741867  41.09879303]
  [ 15.62708759  55.47273254  41.3983078 ]
  [ 17.04044724  54.48454285  40.05719757]]
 [[ 26.25128746  52.36060715  40.28754807]
  [ 27.11155128  52.89438629  40.3939476 ]
  [ 26.49939919  51.40272141  40.58090591]
  ..., 
  [ 16.19880104  54.51780319  41.09930038]
  [ 15.23429489  55.28584671  41.29219437]
  [ 17.12046432  54.87140656  40.28112793]]]...
MDAnalysis.topology.PSF: DEBUG: PSF file /builddir/build/BUILD/MDAnalysis-0.16.1/MDAnalysisTests-0.16.1/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Atomids object at 0xffff585bb090> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Atomnames object at 0xffff585bb0d0> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Atomtypes object at 0xffff585bb110> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Charges object at 0xffff585bb150> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Masses object at 0xffff585bb190> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Resids object at 0xffff585bb210> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Resnums object at 0xffff585bb1d0> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Resnames object at 0xffff585bb250> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Segids object at 0xffff585bb290> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Atomindices object at 0xffff585bb350> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Resindices object at 0xffff585bb390> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Segindices object at 0xffff585bb3d0> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Bonds object at 0xffff585bb450> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Angles object at 0xffff585b5e50> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Dihedrals object at 0xffff585b5cd0> to topology
MDAnalysis.core.universe: DEBUG: _process_attr: Adding <MDAnalysis.core.topologyattrs.Impropers object at 0xffff585b5e10> to topology
MDAnalysis.core.universe: DEBUG: Universe.load_new(): loading [[[ 11.73604393   8.50079727 -10.44528103]
  [ 12.36511898   7.83993578 -10.83484173]
  [ 12.09194851   9.441535   -10.72461128]
  ..., 
  [  6.51260424  18.44701767  -7.13405323]
  [  6.30018616  19.36348534  -7.93591595]
  [  5.58540154  17.5896244   -6.96566153]]
 [[ 12.97540283   8.5564661   -8.79086781]
  [ 13.66876411   8.11343288  -8.09570789]
  [ 13.19270515   9.55035019  -9.03054714]
  ..., 
  [  7.34918356  17.4371357   -6.57104731]
  [  7.43909597  17.90647507  -7.75675392]
  [  6.21367168  17.11502075  -6.09452438]]
 [[ 13.04726887   7.63489389  -9.00248337]
  [ 13.95461655   7.6078577   -8.37104511]
  [ 13.05357552   8.57965088  -9.43249798]
  ..., 
  [  7.36733294  17.43116379  -6.58304739]
  [  7.42952394  18.0872364   -7.64184904]
  [  6.29180861  16.84422874  -6.30289555]]
 ..., 
 [[ 15.64185715   6.52680159  -7.78829479]
  [ 16.17948151   6.46231985  -6.86501741]
  [ 15.8962183    7.40947819  -8.22259808]
  ..., 
  [ 12.37435722  14.83569622  -6.58019352]
  [ 12.4016571   15.75725174  -7.4538641 ]
  [ 11.34606457  14.13633633  -6.49367189]]
 [[ 16.01750946   6.42328167  -8.06725407]
  [ 16.6257       6.14889956  -7.24784851]
  [ 16.26796532   7.37695169  -8.37005711]
  ..., 
  [ 12.80069256  15.77023315  -6.08095074]
  [ 12.70214462  16.68847656  -6.90102482]
  [ 12.32308102  14.67083359  -6.31280994]]
 [[ 16.29778099   6.83979559  -7.62298918]
  [ 16.82201767   6.56630898  -6.70722151]
  [ 16.76083183   7.6656146   -7.95306826]
  ..., 
  [ 12.63667011  15.56686878  -6.11850452]
  [ 12.8277998   16.21443558  -7.16725492]
  [ 11.55105019  14.87915421  -5.94013405]]]...
root: INFO:         Perform pairwise alignment: True
root: INFO:         weighted alignment and RMSD: Mass
root: INFO:         Atoms subset for alignment: name CA
root: INFO:     Calculating similarity matrix . . .
root: INFO:     Done!
root: INFO: Preference -1.00: starting Affinity Propagation
root: INFO: Preference -1.00: converged in 196 iterations
--------------------- >> end captured logging << ---------------------
----------------------------------------------------------------------
Ran 4554 tests in 572.048s
FAILED (SKIP=16, failures=1)
```

NOTE: on big-endian arches (ppc64 and s390x) there are also 67 test errors, but I'll open a separate issue for that.

### Currently version of MDAnalysis:
0.16.1

For the record, this test is failing even on 32 bit x86 in 0.17.0:
```
______ TestEncoreClustering.test_clustering_three_ensembles_two_identical ______
[gw0] linux2 -- Python 2.7.14 /usr/bin/python2

self = <MDAnalysisTests.analysis.test_encore.TestEncoreClustering object at 0xe3f196ec>
ens1 = <Universe with 3341 atoms>, ens2 = <Universe with 3341 atoms>

    def test_clustering_three_ensembles_two_identical(self, ens1, ens2):
        cluster_collection = encore.cluster([ens1, ens2, ens1])
        expected_value = 40
>       assert len(cluster_collection) == expected_value, "Unexpected result:" \
                                                          " {0}".format(cluster_collection)
E       AssertionError: Unexpected result: 0 (size:2,centroid:0): array([ 0, 20])
E         1 (size:2,centroid:1): array([ 1, 21])
E         2 (size:1,centroid:2): array([2])
E         3 (size:1,centroid:3): array([3])
E         4 (size:1,centroid:4): array([4])
E         5 (size:1,centroid:5): array([5])
E         6 (size:1,centroid:6): array([6])
E         7 (size:1,centroid:7): array([7])
E         8 (size:1,centroid:8): array([8])
E         9 (size:1,centroid:9): array([9])
E         10 (size:1,centroid:10): array([10])
E         11 (size:1,centroid:11): array([11])
E         12 (size:1,centroid:12): array([12])
E         13 (size:1,centroid:13): array([13])
E         14 (size:1,centroid:14): array([14])
E         15 (size:1,centroid:15): array([15])
E         16 (size:1,centroid:16): array([16])
E         17 (size:1,centroid:17): array([17])
E         18 (size:4,centroid:18): array([18, 38, 39, 59])
E         19 (size:3,centroid:19): array([19, 40, 60])
E         20 (size:3,centroid:23): array([22, 23, 24])
E         21 (size:2,centroid:26): array([25, 26])
E         22 (size:2,centroid:28): array([27, 28])
E         23 (size:3,centroid:30): array([29, 30, 31])
E         24 (size:3,centroid:33): array([32, 33, 34])
E         25 (size:3,centroid:36): array([35, 36, 37])
E         26 (size:1,centroid:41): array([41])
E         27 (size:1,centroid:42): array([42])
E         28 (size:1,centroid:43): array([43])
E         29 (size:1,centroid:44): array([44])
E         30 (size:1,centroid:45): array([45])
E         31 (size:1,centroid:46): array([46])
E         32 (size:1,centroid:47): array([47])
E         33 (size:1,centroid:48): array([48])
E         34 (size:1,centroid:49): array([49])
E         35 (size:1,centroid:50): array([50])
E         36 (size:1,centroid:51): array([51])
E         37 (size:1,centroid:52): array([52])
E         38 (size:1,centroid:53): array([53])
E         39 (size:1,centroid:54): array([54])
E         40 (size:1,centroid:55): array([55])
E         41 (size:1,centroid:56): array([56])
E         42 (size:1,centroid:57): array([57])
E         43 (size:1,centroid:58): array([58])
E         
E       assert 44 == 40
E        +  where 44 = len(0 (size:2,centroid:0): array([ 0, 20])\n1 (size:2,centroid:1): array([ 1, 21])\n...56])\n42 (size:1,centroid:57): array([57])\n43 (size:1,centroid:58): array([58])\n)

analysis/test_encore.py:422: AssertionError
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -48,7 +48,7 @@ before_install:
   - . "/home/travis/miniconda/etc/profile.d/conda.sh"
   - conda activate base
   - conda update --yes conda
-  - conda install --yes pip pytest==6.1.2 mmtf-python biopython networkx cython matplotlib-base scipy griddataformats hypothesis gsd codecov -c conda-forge -c biobuilds
+  - conda install --yes pip pytest==6.1.2 mmtf-python biopython networkx cython matplotlib-base "scipy>=1.6" griddataformats hypothesis gsd codecov -c conda-forge -c biobuilds
   - pip install pytest-xdist
   - conda info
 install:

--- a/.travis.yml
+++ b/.travis.yml
@@ -48,7 +48,7 @@ before_install:
   - . "/home/travis/miniconda/etc/profile.d/conda.sh"
   - conda activate base
   - conda update --yes conda
-  - conda install pip pytest==6.1.2 mmtf-python biopython networkx cython matplotlib-base scipy griddataformats hypothesis gsd codecov -c conda-forge biobuilds
+  - conda install pip pytest==6.1.2 mmtf-python biopython networkx cython matplotlib-base scipy griddataformats hypothesis gsd codecov -c conda-forge -c biobuilds
   - pip install pytest-xdist
   - conda info
 install:

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,0 +1,54 @@
+version: ~> 1.0
+language: generic
+group: travis_latest
+
+# Only build for develop
+branches:
+  only:
+    - develop
+
+matrix:
+  fast_finish: true
+  include:
+    - python: 3.8
+      name: "Power PC"
+      os: linux
+      arch: ppc64le
+      before_install:
+        - wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-ppc64le.sh -O miniconda.sh
+        - mkdir $HOME/.conda
+        - bash miniconda.sh -b -p $HOME/miniconda
+        - $HOME/miniconda/bin/conda init bash
+        - source ~/.bash_profile
+        - conda activate base
+        - conda update --yes conda
+        - conda install pip pytest==6.1.2 mmtf-python biopython networkx cython matplotlib-base scipy griddataformats hypothesis gsd codecov -c conda-forge biobuilds
+        - pip install pytest-xdist
+        - conda info
+      install:
+        - (cd package/ && python setup.py develop) && (cd testsuite/ && python setup.py install)
+      script:
+        # 4 threads per power9 cores, 2 cores = 8 threads
+        - python -m pytest ./MDAnalysisTests --disable-pytest-warnings -n 8
+
+    - os: linux
+      language: python
+      arch: arm64-graviton2
+      python: 3.8
+      dist: focal
+      virt: vm
+      group: edge
+      before_install:
+        - python -m pip install cython numpy scipy
+        - python -m pip install --no-build-isolation hypothesis matplotlib pytest pytest-cov pytest-xdist tqdm
+      install:
+        - cd package
+        - python setup.py install
+        - cd ../testsuite
+        - python setup.py install
+        - cd ..
+      script:
+        - cd testsuite
+        - python -m pytest ./MDAnalysisTests --disable-pytest-warnings -n 8 -rsx --cov=MDAnalysis
+      after_success:
+        - echo "Override this stage for ARM64"

--- a/.travis.yml
+++ b/.travis.yml
@@ -19,7 +19,7 @@ matrix:
         - mkdir $HOME/.conda
         - bash miniconda.sh -b -p $HOME/miniconda
         - $HOME/miniconda/bin/conda init bash
-        - source ~/.bash_profile
+        - source ~/.bashrc
         - conda activate base
         - conda update --yes conda
         - conda install pip pytest==6.1.2 mmtf-python biopython networkx cython matplotlib-base scipy griddataformats hypothesis gsd codecov -c conda-forge biobuilds
@@ -30,11 +30,14 @@ matrix:
       script:
         # 4 threads per power9 cores, 2 cores = 8 threads
         - python -m pytest ./MDAnalysisTests --disable-pytest-warnings -n 8
+      after_success:
+        - echo "Override this stage for ppc64le"
 
     - os: linux
       language: python
       arch: arm64-graviton2
       python: 3.8
+      name: "ARM64"
       dist: focal
       virt: vm
       group: edge

--- a/.travis.yml
+++ b/.travis.yml
@@ -55,6 +55,6 @@ install:
   - (cd package/ && python setup.py develop) && (cd testsuite/ && python setup.py install)
 script:
   # 4 threads per power9 cores, 2 cores = 8 threads
-  - python -m pytest ./MDAnalysisTests --disable-pytest-warnings -n 8
+  - pytest testsuite/MDAnalysisTests --disable-pytest-warnings -n 8
 after_success:
   - echo "Override this stage for now"

--- a/.travis.yml
+++ b/.travis.yml
@@ -19,8 +19,9 @@ matrix:
         - mkdir $HOME/.conda
         - bash miniconda.sh -b -p $HOME/miniconda
         - $HOME/miniconda/bin/conda init bash
-          #- source ~/.bashrc
-        - source activate base
+        - cat $HOME/.bashrc
+        - source $HOME/.bashrc
+        - conda activate base
         - conda update --yes conda
         - conda install pip pytest==6.1.2 mmtf-python biopython networkx cython matplotlib-base scipy griddataformats hypothesis gsd codecov -c conda-forge biobuilds
         - pip install pytest-xdist
@@ -55,3 +56,4 @@ matrix:
         - python -m pytest ./MDAnalysisTests --disable-pytest-warnings -n 8 -rsx --cov=MDAnalysis
       after_success:
         - echo "Override this stage for ARM64"
+

--- a/.travis.yml
+++ b/.travis.yml
@@ -13,27 +13,7 @@ matrix:
     - python: 3.8
       name: "Power PC"
       os: linux
-      #dist: focal
       arch: ppc64le
-      #before_install:
-      #  - wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-ppc64le.sh -O miniconda.sh
-      #  - mkdir $HOME/.conda
-      #  - bash miniconda.sh -b -p $HOME/miniconda
-      #  - $HOME/miniconda/bin/conda init bash
-      #  - cat $HOME/.bashrc
-      #  - source $HOME/.bashrc
-      #  - conda activate base
-      #  - conda update --yes conda
-      #  - conda install pip pytest==6.1.2 mmtf-python biopython networkx cython matplotlib-base scipy griddataformats hypothesis gsd codecov -c conda-forge biobuilds
-      #  - pip install pytest-xdist
-      #  - conda info
-      #install:
-      #  - (cd package/ && python setup.py develop) && (cd testsuite/ && python setup.py install)
-      #script:
-      #  # 4 threads per power9 cores, 2 cores = 8 threads
-      #  - python -m pytest ./MDAnalysisTests --disable-pytest-warnings -n 8
-      #after_success:
-      #  - echo "Override this stage for ppc64le"
 
     - os: linux
       language: python
@@ -57,6 +37,7 @@ matrix:
         - python -m pytest ./MDAnalysisTests --disable-pytest-warnings -n 8 -rsx --cov=MDAnalysis
       after_success:
         - echo "Override this stage for ARM64"
+
 
 before_install:
   - wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-ppc64le.sh -O miniconda.sh

--- a/.travis.yml
+++ b/.travis.yml
@@ -14,6 +14,24 @@ matrix:
       name: "Power PC"
       os: linux
       arch: ppc64le
+      before_install:
+        - wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-ppc64le.sh -O miniconda.sh
+        - mkdir $HOME/.conda
+        - bash miniconda.sh -b -p $HOME/miniconda
+        - $HOME/miniconda/bin/conda init bash
+        # for some reason the image does not like sourcing .bashrc
+        - . "/home/travis/miniconda/etc/profile.d/conda.sh"
+        - conda activate base
+        - conda update --yes conda
+        - conda install --yes pip pytest==6.1.2 mmtf-python biopython networkx cython matplotlib-base "scipy>=1.6" griddataformats hypothesis gsd codecov -c conda-forge -c biobuilds
+        - pip install pytest-xdist
+        - conda info
+      install:
+        - (cd package/ && python setup.py develop) && (cd testsuite/ && python setup.py install)
+      script:
+        - pytest testsuite/MDAnalysisTests --disable-pytest-warnings -n 2
+      after_sucess:
+        - echo "Override this stage for now"
 
     - os: linux
       language: python
@@ -38,23 +56,3 @@ matrix:
       after_success:
         - echo "Override this stage for ARM64"
 
-
-before_install:
-  - wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-ppc64le.sh -O miniconda.sh
-  - mkdir $HOME/.conda
-  - bash miniconda.sh -b -p $HOME/miniconda
-  - $HOME/miniconda/bin/conda init bash
-  - cat $HOME/.bashrc
-  - . "/home/travis/miniconda/etc/profile.d/conda.sh"
-  - conda activate base
-  - conda update --yes conda
-  - conda install --yes pip pytest==6.1.2 mmtf-python biopython networkx cython matplotlib-base "scipy>=1.6" griddataformats hypothesis gsd codecov -c conda-forge -c biobuilds
-  - pip install pytest-xdist
-  - conda info
-install:
-  - (cd package/ && python setup.py develop) && (cd testsuite/ && python setup.py install)
-script:
-  # 4 threads per power9 cores, 2 cores = 8 threads
-  - pytest testsuite/MDAnalysisTests --disable-pytest-warnings -n 8
-after_success:
-  - echo "Override this stage for now"

--- a/.travis.yml
+++ b/.travis.yml
@@ -19,8 +19,8 @@ matrix:
         - mkdir $HOME/.conda
         - bash miniconda.sh -b -p $HOME/miniconda
         - $HOME/miniconda/bin/conda init bash
-        - source ~/.bashrc
-        - conda activate base
+          #- source ~/.bashrc
+        - source activate base
         - conda update --yes conda
         - conda install pip pytest==6.1.2 mmtf-python biopython networkx cython matplotlib-base scipy griddataformats hypothesis gsd codecov -c conda-forge biobuilds
         - pip install pytest-xdist

--- a/.travis.yml
+++ b/.travis.yml
@@ -48,7 +48,7 @@ before_install:
   - . "/home/travis/miniconda/etc/profile.d/conda.sh"
   - conda activate base
   - conda update --yes conda
-  - conda install pip pytest==6.1.2 mmtf-python biopython networkx cython matplotlib-base scipy griddataformats hypothesis gsd codecov -c conda-forge -c biobuilds
+  - conda install --yes pip pytest==6.1.2 mmtf-python biopython networkx cython matplotlib-base scipy griddataformats hypothesis gsd codecov -c conda-forge -c biobuilds
   - pip install pytest-xdist
   - conda info
 install:

--- a/.travis.yml
+++ b/.travis.yml
@@ -45,7 +45,7 @@ before_install:
   - bash miniconda.sh -b -p $HOME/miniconda
   - $HOME/miniconda/bin/conda init bash
   - cat $HOME/.bashrc
-  - source $HOME/.bashrc
+  - . "/home/travis/miniconda/etc/profile.d/conda.sh"
   - conda activate base
   - conda update --yes conda
   - conda install pip pytest==6.1.2 mmtf-python biopython networkx cython matplotlib-base scipy griddataformats hypothesis gsd codecov -c conda-forge biobuilds

--- a/.travis.yml
+++ b/.travis.yml
@@ -13,26 +13,27 @@ matrix:
     - python: 3.8
       name: "Power PC"
       os: linux
+      dist: focal
       arch: ppc64le
-      before_install:
-        - wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-ppc64le.sh -O miniconda.sh
-        - mkdir $HOME/.conda
-        - bash miniconda.sh -b -p $HOME/miniconda
-        - $HOME/miniconda/bin/conda init bash
-        - cat $HOME/.bashrc
-        - source $HOME/.bashrc
-        - conda activate base
-        - conda update --yes conda
-        - conda install pip pytest==6.1.2 mmtf-python biopython networkx cython matplotlib-base scipy griddataformats hypothesis gsd codecov -c conda-forge biobuilds
-        - pip install pytest-xdist
-        - conda info
-      install:
-        - (cd package/ && python setup.py develop) && (cd testsuite/ && python setup.py install)
-      script:
-        # 4 threads per power9 cores, 2 cores = 8 threads
-        - python -m pytest ./MDAnalysisTests --disable-pytest-warnings -n 8
-      after_success:
-        - echo "Override this stage for ppc64le"
+      #before_install:
+      #  - wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-ppc64le.sh -O miniconda.sh
+      #  - mkdir $HOME/.conda
+      #  - bash miniconda.sh -b -p $HOME/miniconda
+      #  - $HOME/miniconda/bin/conda init bash
+      #  - cat $HOME/.bashrc
+      #  - source $HOME/.bashrc
+      #  - conda activate base
+      #  - conda update --yes conda
+      #  - conda install pip pytest==6.1.2 mmtf-python biopython networkx cython matplotlib-base scipy griddataformats hypothesis gsd codecov -c conda-forge biobuilds
+      #  - pip install pytest-xdist
+      #  - conda info
+      #install:
+      #  - (cd package/ && python setup.py develop) && (cd testsuite/ && python setup.py install)
+      #script:
+      #  # 4 threads per power9 cores, 2 cores = 8 threads
+      #  - python -m pytest ./MDAnalysisTests --disable-pytest-warnings -n 8
+      #after_success:
+      #  - echo "Override this stage for ppc64le"
 
     - os: linux
       language: python
@@ -57,3 +58,22 @@ matrix:
       after_success:
         - echo "Override this stage for ARM64"
 
+before_install:
+  - wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-ppc64le.sh -O miniconda.sh
+  - mkdir $HOME/.conda
+  - bash miniconda.sh -b -p $HOME/miniconda
+  - $HOME/miniconda/bin/conda init bash
+  - cat $HOME/.bashrc
+  - source $HOME/.bashrc
+  - conda activate base
+  - conda update --yes conda
+  - conda install pip pytest==6.1.2 mmtf-python biopython networkx cython matplotlib-base scipy griddataformats hypothesis gsd codecov -c conda-forge biobuilds
+  - pip install pytest-xdist
+  - conda info
+install:
+  - (cd package/ && python setup.py develop) && (cd testsuite/ && python setup.py install)
+script:
+  # 4 threads per power9 cores, 2 cores = 8 threads
+  - python -m pytest ./MDAnalysisTests --disable-pytest-warnings -n 8
+after_success:
+  - echo "Override this stage for now"

--- a/.travis.yml
+++ b/.travis.yml
@@ -14,6 +14,7 @@ matrix:
       name: "Power PC"
       os: linux
       arch: ppc64le
+      if: type = cron
       before_install:
         - wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-ppc64le.sh -O miniconda.sh
         - mkdir $HOME/.conda
@@ -41,6 +42,7 @@ matrix:
       dist: focal
       virt: vm
       group: edge
+      if: type = cron
       before_install:
         - python -m pip install cython numpy scipy
         - python -m pip install --no-build-isolation hypothesis matplotlib pytest pytest-cov pytest-xdist tqdm

--- a/.travis.yml
+++ b/.travis.yml
@@ -13,7 +13,7 @@ matrix:
     - python: 3.8
       name: "Power PC"
       os: linux
-      dist: focal
+      #dist: focal
       arch: ppc64le
       #before_install:
       #  - wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-ppc64le.sh -O miniconda.sh

--- a/package/CHANGELOG
+++ b/package/CHANGELOG
@@ -134,7 +134,7 @@ Enhancements
   
 Changes
   * Introduces encore specific C compiler arguments to allow for lowering of
-    optimisations on non-x86 platforms (PR #3149)
+    optimisations on non-x86 platforms (Issue #1389, PR #3149)
   * Continuous integration uses mamba rather than conda to install the
     dependencies (PR #2983)
   * removes deprecated `as_Universe` function from MDAnalysis.core.universe,

--- a/package/CHANGELOG
+++ b/package/CHANGELOG
@@ -84,6 +84,8 @@ Fixes
   * Fix syntax warning over comparison of literals using is (Issue #3066)
 
 Enhancements
+  * Adds preliminary support for the ppc64le platform with minimal
+    dependencies (Issue #3127, PR #3149)
   * Caches can now undergo central validation at the Universe level, opening
     the door to more useful caching. Already applied to fragment caching
     (Issue #2376, PR #3135)
@@ -131,6 +133,8 @@ Enhancements
     explicit in the topology (Issue #2468, PR #2775)
   
 Changes
+  * Introduces encore specific C compiler arguments to allow for lowering of
+    optimisations on non-x86 platforms (PR #3149)
   * Continuous integration uses mamba rather than conda to install the
     dependencies (PR #2983)
   * removes deprecated `as_Universe` function from MDAnalysis.core.universe,

--- a/package/setup.py
+++ b/package/setup.py
@@ -261,15 +261,8 @@ def extensions(config):
     use_cython = config.get('use_cython', default=not is_release)
     use_openmp = config.get('use_openmp', default=True)
 
-    if platform.machine() == 'aarch64':
-        # reduce optimization level for ARM64 machines
-        # because of issues with test failures sourcing to:
-        # MDAnalysis/analysis/encore/clustering/src/ap.c
-        extra_compile_args = ['-std=c99', '-ffast-math', '-O1', '-funroll-loops',
-                              '-fsigned-zeros']
-    else:
-        extra_compile_args = ['-std=c99', '-ffast-math', '-O3', '-funroll-loops',
-                              '-fsigned-zeros']  # see #2722
+    extra_compile_args = ['-std=c99', '-ffast-math', '-O3', '-funroll-loops',
+                          '-fsigned-zeros']  # see #2722
     define_macros = []
     if config.get('debug_cflags', default=False):
         extra_compile_args.extend(['-Wall', '-pedantic'])
@@ -280,6 +273,15 @@ def extensions(config):
     arch = config.get('march', default=False)
     if arch:
         extra_compile_args.append('-march={}'.format(arch))
+
+    # encore is sensitive to floating point accuracy, especially on non-x86
+    # to avoid reducing optimisations on everything, we make a set of compile
+    # args specific to encore see #2997 for an example of this.
+    encore_compile_args = [a for a in extra_compile_args if 'O3' not in a]
+    if platform.machine() == 'aarch64' or platform.machine() == 'ppc64le':
+        encore_compile_args.append('-O1')
+    else:
+        encore_compile_args.append('-O3')
 
     cpp_extra_compile_args = [a for a in extra_compile_args if 'std' not in a]
     cpp_extra_compile_args.append('-std=c++11')
@@ -401,21 +403,21 @@ def extensions(config):
                                 sources=['MDAnalysis/analysis/encore/cutils' + source_suffix],
                                 include_dirs=include_dirs,
                                 define_macros=define_macros,
-                                extra_compile_args=extra_compile_args)
+                                extra_compile_args=encore_compile_args)
     ap_clustering = MDAExtension('MDAnalysis.analysis.encore.clustering.affinityprop',
                                  sources=['MDAnalysis/analysis/encore/clustering/affinityprop' + source_suffix,
                                           'MDAnalysis/analysis/encore/clustering/src/ap.c'],
                                  include_dirs=include_dirs+['MDAnalysis/analysis/encore/clustering/include'],
                                  libraries=mathlib,
                                  define_macros=define_macros,
-                                 extra_compile_args=extra_compile_args)
+                                 extra_compile_args=encore_compile_args)
     spe_dimred = MDAExtension('MDAnalysis.analysis.encore.dimensionality_reduction.stochasticproxembed',
                               sources=['MDAnalysis/analysis/encore/dimensionality_reduction/stochasticproxembed' + source_suffix,
                                        'MDAnalysis/analysis/encore/dimensionality_reduction/src/spe.c'],
                               include_dirs=include_dirs+['MDAnalysis/analysis/encore/dimensionality_reduction/include'],
                               libraries=mathlib,
                               define_macros=define_macros,
-                              extra_compile_args=extra_compile_args)
+                              extra_compile_args=encore_compile_args)
     nsgrid = MDAExtension('MDAnalysis.lib.nsgrid',
                              ['MDAnalysis/lib/nsgrid' + cpp_source_suffix],
                              include_dirs=include_dirs,

